static/preview.html

```diff
@@ -551,6 +551,53 @@
     <script src="/@/artifacts/index.js"></script>
 
     <script>
+      function mockStorage( { toLocalStorageKey = '' }) {
+        let storage = {};
+
+        function update(){
+          if (toLocalStorageKey !== ''){
+            localStorage.setItem(toLocalStorageKey, JSON.stringify(storage))
+          }
+        }
+        
+        if (toLocalStorageKey !== ''){
+          storage = JSON.parse(localStorage.getItem(toLocalStorageKey) || '{}');
+        }
+
+        return {
+          setItem: function(key, value) {
+            return new Promise(resolve => {
+              storage[key] = value || '';
+              update();
+              resolve();
+            });
+          },
+          getItem: function(key) {
+            return new Promise(resolve => {
+              resolve(key in storage ? storage[key] : null);
+            });
+          },
+          removeItem: function(key) {
+            return new Promise(resolve => {
+              delete storage[key];
+              update();
+              resolve()
+            });
```

Comment:
This is another alternative implementation of the same code, without callbacks. No need to change it, just informative

```suggestion
            delete storage[key];
            update();
            return Promise.resolve()
```

Reply:
No need, but cleaner